Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1394,6 +1394,78 @@ void RenderBox::clearOverridingLogicalWidthLength()
         gOverridingLogicalWidthLengthMap->remove(this);
 }
 
+std::optional<FlowRelativeDirection> RenderBox::physicalToFlowRelativeDirectionMapping(PhysicalDirection direction, LayoutCalculationPhase layoutCalculationPhase) const
+{
+    // FIXME: Return type should be just FlowRelativeDirection once all code paths have been implemented
+    auto isHorizontalWritingMode = style().isHorizontalWritingMode();
+    auto isLeftToRightDirection = style().isLeftToRightDirection();
+    // vertical-rl and horizontal-bt writing modes
+    auto isFlippedBlocksWritingMode = style().isFlippedBlocksWritingMode();
+    if (layoutCalculationPhase == LayoutCalculationPhase::Positioning) {
```

Comment:
/* Unofficial Review */
Using an early return, we can mitigate the if depth.

```
if (layoutCalculationPhaase != LayoutCalculationPhase::Positioning) {
    ASSERT_NOT_IMPLEMENTED_YET();
    return {}; // std::nullopt?
}

auto isHorizontalWritingMode...;
...

if (isHorizontalWritingMode == containingBlock()->style().isHorizontalWritingMode()) {
    switch (direction) {
...
}
```

Reply:
We ended up not including LayoutCalculationPhase as part of https://github.com/WebKit/WebKit/pull/11640